sdk/core/src/sleep.rs

```diff
@@ -25,7 +25,7 @@ impl Future for Sleep {
             let waker = cx.waker().clone();
             let duration = self.duration;
             self.get_mut().thread = Some(thread::spawn(move || {
-                thread::sleep(duration);
+                thread::sleep(duration.try_into().unwrap_or_default());
```

Comment:
Is this because the std `Duration` is not signed? I'm not sure I like this... Might be best to keep using the std library's duration in this case?

Reply:
[time::Duration](https://docs.rs/time/latest/time/struct.Duration.html) uses a i64 of seconds and `std::time::Duration` uses an u64 of seconds to allow for negative durations. It felt a bit more consistent to switch to `time::Duration`. The API felt a bit nicer as well, but we could use `std::time::Duration`.

Reply:
I migrated to `std::time::Duration`.